### DISCUSSION TITLE
Is directus secure?

### DISCUSSION THREAD

**Discussion by tulindor** [directus/directus#2875] [Archive]
Im new to this CMS and i love the fact that is PHP based, so it can run on mostly any server, making it more approachable in pricing terms.

But there is a big question in retrieving data from the API url, de documentation to consult the API is not to clear, and i also found some query params that shows way more information that i wish any user could se... 

Example:
I've got a home item, where im gonna show the information that builds the home page, lets say the url is:
http://someurl.com/pagename/items/home

That its on public level, so any one that enter the front-end this data can be shown without a login or else. So far So good.

But if the same url i add the next parameter
http://directus.test/test/items/home?fields=*.*.*

The endpoint shows sensitive user information such as email and even worst user token... and anyone can access that data.

I think this is a big fail or maybe im not doing the right aproach to the endpoints use, i hope some one can shed some light on the matter.

Sorry if the english is a little bad! it's not my main language.

**Comment by rijkvanzanten** [comment]
> and anyone can access that data.

That totally depends on how you configure your permissions. If you give access to the public role to everything, everything will be.. public.

By default, all data in Directus is private, so only things you open up through permissions are publicly accessible.

**Reply by tulindor** [reply]
Ok, following this i've mannaged to see that there are fileds that i can show or hide to the api. I hide those, but keep showing, then I tried the same endpoint with a user not logged in and now it shows less info, i think that answer my question.

**Reply by rijkvanzanten** [reply]
Yeah! Your user is most likely an admin user, which has full access to everything.
